Framework/API/inc/MantidAPI/ComponentInfo.h

```diff
@@ -59,10 +59,21 @@ class MANTID_API_DLL ComponentInfo {
       const Mantid::Beamline::ComponentInfo &componentInfo,
       boost::shared_ptr<const std::vector<Mantid::Geometry::IComponent *>>
           componentIds);
+
+  ComponentInfo(
+      const Mantid::Beamline::ComponentInfo &componentInfo,
+      boost::shared_ptr<const std::vector<Mantid::Geometry::IComponent *>>
+          componentIds,
+  boost::shared_ptr<std::unordered_map<Geometry::IComponent *, size_t>>
```

Comment:
Maybe this could be made `const std::unordered_map`, to prevent accidential changes.